homeassistant/components/tellduslive/config_flow.py

```diff
@@ -56,38 +63,35 @@ def _get_auth_url(self):
 
     async def async_step_auth(self, user_input=None):
         """Handle the submitted configuration."""
-        if not self._session:
-            from tellduslive import Session
-            self._session = Session(
-                public_key=PUBLIC_KEY,
-                private_key=NOT_SO_PRIVATE_KEY,
-                host=self._host,
-                application=APPLICATION_NAME,
-            )
-
-        if user_input is not None and self._session.authorize():
-            host = self._host or CLOUD_NAME
-            if self._host:
-                session = {
-                    KEY_HOST: host,
-                    KEY_TOKEN: self._session.access_token
-                }
+        errors = {}
+        if user_input is not None:
+            if self._session.authorize():
```

Comment:
This is doing I/O correct? Then we need to run it in the executor thread pool, via `hass.async_add_executor_job`.

Reply:
True!

We must have missed it the last time 😃 .